### PR TITLE
FEATURE: Add nodemigrations command to list all available node migration

### PR DESCRIPTION
**Pull request by dlubitz**
Adds a cli command to list all available node migrations.

`nodemigration:list`

This was in previous versions built-in in the `node:migrationstatus`, but as we will not have this command in 9.0, this shall at least provide an overview over available node migrations.

```
> ./flow nodemigration:list

Available migrations

+----------------+---------------------+-----------+-------------------------------+
| Version        | Date                | Package   | Comments                      |
+----------------+---------------------+-----------+-------------------------------+
| 20231013141245 | 13-10-2023 14:12:45 | Neos.Demo | This is a important migration |
+----------------+---------------------+-----------+-------------------------------+
```

Related to: #4571

### CODE REVIEW

File: Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php [hunk 1]
```diff
@@ -114,6 +114,40 @@ public function kickstartCommand(string $packageKey): void
        $this->outputLine('Your node migration has been created successfully.');
     }
 
+    /**
+     * List available migrations
+     *
+     * @return void
```
Comment (bwaidelich) [inline Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php]: mini nitpick, the return type annotation is redundant:
```suggestion
```

**Review by ahaeslich — APPROVED** [review]

File: Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php [hunk 2]
```diff
@@ -114,6 +114,40 @@ public function kickstartCommand(string $packageKey): void
...
+     * @see neos.contentrepositoryregistry:nodemigration:list
+     */
+    public function listCommand(): void
+    {
+        $this->outputLine();
```
Comment (bwaidelich) [inline Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php]: We don't typically add an empty line before CLI output, why did you add this?

**Review by bwaidelich — APPROVED** [review]
I like. Just two nitpick you are free to ignore :)

File: Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php (hunk 2 above)
Comment (dlubitz) [reply Neos.ContentRepositoryRegistry/Classes/Command/NodeMigrationCommandController.php]: 🤷‍♂️ It's copy paste code. I didn't question it. Will remove it.
